src/docs/development/platform-integration/add-to-app/performance.md

```diff
@@ -0,0 +1,174 @@
+on **Android** or via [`FlutterViewController initWithProject: nibName: bundle:`]({{site.api}}/objcdoc/Classes/FlutterViewController.html#/c:objc(cs)FlutterViewController(im)initWithProject:nibName:bundle:)
+on **iOS**. An implicit `FlutterEngine` is created in these cases. ;
+
+Behind the scene, both platform's UI components serves to provide the
+`FlutterEngine` with a rendering surface such as a [Surface](https://developer.android.com/reference/android/view/Surface)
+on **Android** or a [CAEAGLLayer](https://developer.apple.com/documentation/quartzcore/caeagllayer)
+or [CAMetalLayer](https://developer.apple.com/documentation/quartzcore/cametallayer)
+on **iOS**.
+
+At this point, the [Layer]({{site.api}}/flutter/rendering/Layer-class.html)
+tree generated by your Flutter program per frame will be converted into
+OpenGL (or Vulkan or Metal) GPU instructions.
+
+## Memory and latency
+
+Showing a Flutter UI has non-trivial latency cost. This cost can be alleviated
+by starting the Flutter engine ahead of time.
+
+The most relevant choice for users in add-to-app scenarios is to decide
+when to pre-load a `FlutterEngine` (that is, to load the Flutter library,
+start the Dart VM and run entrypoint in an Isolate) and the memory and latency
+cost of that pre-warm. And how it affects the memory and latency cost of
+rendering a first Flutter frame when UI is subsequently attached to that
+`FlutterEngine`.
+
+As of September of 2019, testing on a low-end 2015 class device in release-AOT
+mode, pre-warming the `FlutterEngine` costs:
+
+- 42MB and 1530ms to prewarm on **Android**. 330ms of it is a blocking call on
+the main thread.
+- 22MB and 860ms to prewarm on **iOS**. 260ms of it is a blocking call on the
+main thread.
+
+A Flutter UI can be attached during the pre-warm. The remaining time would be
+joined to the time-to-first-frame latency.
+
+Memory-wise, a cost sample (variable depending on use case) could be:
+
+- ~4MB OS's memory usage for creating pthreads.
+- ~10MB GPU driver memory.
+- ~1MB for Dart runtime-managed memory.
+- ~5MB for Dart-loaded font maps.
+
+Latency-wise, a cost sample (variable depending on use case) could be:
+
+- ~20ms to collect the Flutter assets from the application package.
+- ~15ms to dlopen the Flutter engine library.
+- ~200ms to create the Dart VM, load the AOT snapshot.
+- ~200ms to load Flutter dependent fonts and assets.
+- ~400ms to run the entrypoint, create the first widget tree, compile the needed
+GPU shader programs.
+
+The `FlutterEngine` should be pre-warmed late enough such as to delay the
+memory consumption needed but also early enough to avoid joining the
+Flutter engine start-up time into the first frame latency of showing Flutter.
+
+The exact timing depends on the app's structure and heuristic. An example would
+be to load the Flutter engine in the screen before the screen drawn by Flutter.
```

Comment:
before the screen drawn => before Flutter draws the screen

Reply:
the thing that's before is a position (the other screen), rather than a time (when Flutter draws)